system-parachains/coretime/coretime-polkadot/src/genesis_config_presets.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Genesis configs presets for the Polkadot Coretime runtime
+
+use crate::*;
+use sp_std::vec::Vec;
+use system_parachains_constants::genesis_presets::*;
+
+const CORETIME_POLKADOT_ED: Balance = ExistentialDeposit::get();
+
+fn coretime_polkadot_genesis(
+	invulnerables: Vec<(AccountId, AuraId)>,
+	endowed_accounts: Vec<AccountId>,
+	id: ParaId,
+) -> serde_json::Value {
+	serde_json::json!({
```

Comment:
Not sure why we not use the `GenesisConfig` type from the runtime here.

Reply:
Not sure what you mean here

Reply:
Instead of writing here a `json`, you import `super::GenesisConfig`.

Reply:
Ah found it: `RuntimeGenesisConfig`, had to cargo expand to find it. Thanks for the tip! Maybe it's [this comment](https://github.com/seadanda/runtimes/blob/donal-coretime-polkadot/system-parachains/coretime/coretime-polkadot/src/genesis_config_presets.rs#L63) in the `genesis_config_presets.rs` for all runtimes that says it will panic if we pass anything as the `aura` key, and using the struct means we'll have to pass something for all keys. Maybe @michalkucharczyk has more context

Reply:
Naively added on a branch https://github.com/seadanda/runtimes/commit/b552377d66d7f0a5d466cc843d0ee6edba880016, but would like to hear about this aura panic.

Reply:
If you use rust struct here, then the function will return the _json blob_ containing all the fields of the `RuntimeGenesisConfig`, some of them set to default.

Having json here will limit the number of keys in the preset (patch) to only those which actually are changed.

I used _json_ everywhere because I wanted preset to be as compact as possible.

On the other hand using rust struct should guarantee that json is correct. In case of using json this verification can only be done during execution (when json is deserialized). But this shall be covered in tests.

Reply:
Conversation can continue on https://github.com/polkadot-fellows/runtimes/issues/431